src/motor_hardware.cc

```diff
@@ -142,6 +144,27 @@ void MotorHardware::readInputs() {
                     }
                     break;
                 }
+                case MotorMessage::REG_BATTERY_VOLTAGE: {
+                    int32_t data = mm.getData();
+                    sensor_msgs::BatteryState bstate;
+                    bstate.voltage = (float)data * pid_params.battery_voltage_multiplier +
+                                   pid_params.battery_voltage_offset;
+                    bstate.current = std::numeric_limits<float>::quiet_NaN();
+                    bstate.charge = std::numeric_limits<float>::quiet_NaN();
+                    bstate.capacity = std::numeric_limits<float>::quiet_NaN();
+                    bstate.design_capacity = std::numeric_limits<float>::quiet_NaN();
+                    if (bstate.voltage < 20.0) 
+                      bstate.percentage = 0.0;
```

Comment:
Why use an if, when you could just use std::max(0.0, bstate.percentage), it would be cleaner I think.